lib/liquid/combinators/general.ex

```diff
@@ -3,6 +3,14 @@ defmodule Liquid.Combinators.General do
   General purpose combinators used by almost every other combinator
   """
   import NimbleParsec
+  alias Liquid.Combinators.LexicalToken
+
+  @type comparison_operators :: :== | :!= | :> | :< | :>= | :<= | :contains
+  @type condition :: [
+          condition: {LexicalToken.value(), comparison_operators(), LexicalToken.value()}
+        ]
+  @type liquid_variable :: [liquid_variable: LexicalToken.variable_value(), filters: filter()]
+  @type filter :: [filter: String.t(), params: [values: LexicalToken.value()]]
```

Comment:
Whats the added value for adding a `values` keyword/`{:values, []}` tuples if the entire list contains only LexicalToken values?

Reply:
Every part in the AST have a tag. We can optimize the representation without tags but then we need the context and it is harder. This also makes it consistent with the rest of the AST. We do not want to have separate lists without tags. However, there is an error, it is _value:_, not _values:_

Corrected.